### PR TITLE
Migrate lccs get snapshot data

### PR DESCRIPTION
**Pull request by mukhtyar**
This PR includes adding functionality to fetch data and calculate metrics for the Snapshot table in LCCS tool.

### CODE REVIEW

**Comment by mukhtyar** [comment]
[LCCS: Add data fetching and transformation](https://trello.com/c/ZwCSHzG2/171-lccs-add-data-fetching-and-transformation)

File: src/routes/tools/local-climate-change-snapshot/_data.js [hunk 1]
```diff
@@ -132,10 +135,15 @@ const createRanges = (_data) => {
  * @param {string} method - default is GET, POST for uploaded boundaries
  * @return {array}
  */
-export async function getObserved(config, params, method = "GET") {
+export async function getObserved(
+  config,
+  params,
+  method = "GET",
```
Comment (clhenrick) [inline src/routes/tools/local-climate-change-snapshot/_data.js:142]: if the `method` param is the least likely to be modified I might make that the last param as it's optional. Or wrap the params in an object.

Comment (clhenrick) [reply src/routes/tools/local-climate-change-snapshot/_data.js:142]: Don't forget to update the doc string comment too with the param for `searchStr`

File: src/routes/tools/local-climate-change-snapshot/_data.js [hunk 2]
```diff
@@ -158,10 +166,15 @@ export async function getObserved(config, params, method = "GET") {
  * @param {string} method - default is GET, POST for uploaded boundaries
  * @return {array}
  */
-export async function getProjections(config, params, method = "GET") {
+export async function getProjections(
+  config,
+  params,
+  method = "GET",
```
Comment (clhenrick) [inline src/routes/tools/local-climate-change-snapshot/_data.js:174]: Ditto here for the `method` param and updating the doc string comment.

File: src/routes/tools/local-climate-change-snapshot/_data.js [hunk 3]
```diff
@@ -200,3 +221,60 @@ export function getQueryParams({ location, boundary, imperial = true, stat }) {
...
+      );
+      let avg = null;
+      if (filteredValues.length) {
+        avg = mean(filteredValues, (d) => d.value);
+      }
+      return { periodId, periodLabel, scenarioId, scenarioLabel, avg };
+    });
+    return [...dataByPeriods];
+  });
+  return merge(data).filter(({ avg }) => avg);
```
Comment (clhenrick) [inline src/routes/tools/local-climate-change-snapshot/_data.js]: if `avg` is ever `0` here then those values will be filtered out, is that intended?

File: src/routes/tools/local-climate-change-snapshot/_data.js [hunk 4]
```diff
@@ -200,3 +221,60 @@ export function getQueryParams({ location, boundary, imperial = true, stat }) {
...
+      let max = null;
+      if (filteredValues.length && filteredValues.length === 1) {
+        min = filteredValues[0].min;
+        max = filteredValues[0].max;
+      }
+      return { periodId, periodLabel, scenarioId, scenarioLabel, min, max };
+    });
+    return [...dataByPeriods];
+  });
+  return merge(data).filter(({ min, max }) => min && max);
```
Comment (clhenrick) [inline src/routes/tools/local-climate-change-snapshot/_data.js]: if `min` or `max` is ever `0` here then those values will be filtered out, is that intended?

**Review by clhenrick — COMMENTED** [review]
Looks good, just some small suggestions / questions.

File: src/routes/tools/local-climate-change-snapshot/explore.svelte [hunk 5]
```diff
@@ -85,18 +100,27 @@
         ),
       };
       // Get params object for querying the Cal-Adapt API
+      // extra months param required for April SWE indicator
+      const months = $indicatorStore.id === "swe" ? DEFAULT_SWE_MONTH : null;
```
Comment (clhenrick) [inline src/routes/tools/local-climate-change-snapshot/explore.svelte:104]: @mukhtyar when looking at SWE, will the user be able to select the months? Or do they always default to the default SWE months?

Comment (mukhtyar) [reply src/routes/tools/local-climate-change-snapshot/explore.svelte:104]: No, the user will not select months. The indicator is for April SWE only.

File: src/routes/tools/local-climate-change-snapshot/_data.js (hunk 1 above)
Comment (mukhtyar) [reply src/routes/tools/local-climate-change-snapshot/_data.js:142]: Fixed in https://github.com/berkeley-gif/caladapt-website-2021/pull/209/commits/a0404de4c1dba0b67f5b07113c31c051e6c2f015

File: src/routes/tools/local-climate-change-snapshot/_data.js (hunk 2 above)
Comment (mukhtyar) [reply src/routes/tools/local-climate-change-snapshot/_data.js:174]: Fixed in https://github.com/berkeley-gif/caladapt-website-2021/pull/209/commits/a0404de4c1dba0b67f5b07113c31c051e6c2f015

File: src/routes/tools/local-climate-change-snapshot/_data.js (hunk 4 above)
Comment (mukhtyar) [reply src/routes/tools/local-climate-change-snapshot/_data.js]: Good catch. `0` is a valid value. Added a strict check for `null` in https://github.com/berkeley-gif/caladapt-website-2021/pull/209/commits/a0819d5619e39c5361059fc4870a8fe185e4b9fc

**Pull request merged**